libraries/ms-common/test/test_delay.c

```diff
@@ -0,0 +1,26 @@
+#include "delay.h"
+
+#include "interrupt.h"
+#include "soft_timer.h"
+#include "unity.h"
+
+// These tests serve a dual purpose as they also implicitly test the wait module.
+
+void setup_test(void) {
+  interrupt_init();
+  soft_timer_init();
+}
+
+void teardown_test(void) {}
+
+void test_delay_us(void) {
+  delay_us(10000);
+}
+
+void test_delay_ms(void) {
+  delay_ms(10);
+}
+
+void test_delay_s(void) {
+  delay_s(1);
```

Comment:
They are more of a sanity check that this is working and the time periods are correct. I removed the longer ones. However, I feel keeping at least one is important as it serves to validate that after calling `wait()` the MCU wakes up on interrupts which I think is valuable.